main.tf

```diff
@@ -39,10 +39,24 @@ resource "aws_route53_zone" "zone" {
 locals {
   records_expanded = [
     for record in var.records : merge({
-      record_id = join("-", compact([lower(record.type), lower(record.name)]))
-      ttl       = null
-      records   = null
-      alias     = null
+      # if no name is set for the record, we use the zones name
+      record_id = join("-", compact([lower(record.type), try(lower(record.name), ""),
+        try(
+          "${lower(record.set_identifier)}-${lower(record.weight)}",                                    # weighted routing policy
+          "${lower(record.set_identifier)}-${lower(record.failover)}-${lower(record.health_check_id)}", # failover
```

Comment:
also `set_identifier` will do the job i guess it's not about naming resources by all variables they can have but giving it a uniq name...see other comment